example/src/new_api/swipeable/Swipeable.tsx

```diff
@@ -0,0 +1,525 @@
+        },
+        reset() {
+          dragX.value = 0;
+          transX.value = 0;
+          rowState.value = 0;
+        },
+      }),
+      []
+    );
+
+    const defaultProps = {
+      friction: 1,
+      overshootFriction: 1,
+    };
+
+    const {
+      friction = defaultProps.friction,
+      overshootFriction = defaultProps.overshootFriction,
+    } = props;
+
+    const transX = useSharedValue(0);
+    const showLeftAction = useSharedValue(0);
+    const leftActionTranslate = useSharedValue(0);
+    const showRightAction = useSharedValue(0);
+    const rightActionTranslate = useSharedValue(0);
+    const composedX = useDerivedValue(() => rowTranslation.value + dragX.value);
+
+    const currentOffset = () => {
+      'worklet';
+      const rightWidth = rowWidth.value - rightOffset.value;
+      if (rowState.value === 1) {
+        return leftWidth.value;
+      } else if (rowState.value === -1) {
+        return -rightWidth;
+      }
+      return 0;
+    };
+
+    const updateAnimatedEvent = () => {
+      'worklet';
+      const rightWidth = Math.max(0, rowWidth.value - rightOffset.value);
+
+      const {
+        overshootLeft = leftWidth.value > 0,
+        overshootRight = rightWidth > 0,
+      } = props;
+
+      dragX.value = interpolate(dragX.value, [0, friction!], [0, 1]);
+
+      transX.value = interpolate(
+        composedX.value,
+        [-rightWidth - 1, -rightWidth, leftWidth.value, leftWidth.value + 1],
+        [
+          -rightWidth - (overshootRight ? 1 / overshootFriction! : 0),
+          -rightWidth,
+          leftWidth.value,
+          leftWidth.value + (overshootLeft ? 1 / overshootFriction! : 0),
+        ]
+      );
+
+      showLeftAction.value =
+        leftWidth.value > 0
+          ? interpolate(transX.value, [-1, 0, leftWidth.value], [0, 0, 1])
+          : 0;
+      leftActionTranslate.value = interpolate(
+        showLeftAction.value,
+        [0, Number.MIN_VALUE],
+        [-10000, 0],
+        Extrapolation.CLAMP
+      );
+      showRightAction.value =
+        rightWidth > 0
+          ? interpolate(transX.value, [-rightWidth, 0, 1], [1, 0, 0])
+          : 0;
+      rightActionTranslate.value = interpolate(
+        showRightAction.value,
+        [0, Number.MIN_VALUE],
+        [-10000, 0],
+        Extrapolation.CLAMP
+      );
+    };
+
+    const animateRow = (
+      fromValue: number,
+      toValue: number,
+      velocityX?: number
+    ) => {
+      'worklet';
+      dragX.value = 0;
+      transX.value = fromValue;
+
+      rowState.value = Math.sign(toValue);
+
+      transX.value = withSpring(
+        toValue,
+        {
+          duration: 1000,
+          dampingRatio: 1.2,
+          stiffness: 500,
+          velocity: velocityX,
+          ...props.animationOptions,
+        },
+        (isFinished) => {
+          if (isFinished) {
+            if (toValue > 0) {
+              runOnJS(() => props.onSwipeableOpen?.('left', swipeableMethods));
```

Comment:
I have to ask you whether you (c.c. @j-piasecki) think these frequent uses of `runOnJS` are ok, or if you'd prefer to have them replaced with something else.
I've had to place them to avoid crashes on android, which otherwise required every callback to be workletized.